### PR TITLE
support group by with 2 parameters, add tolist

### PR DESCRIPTION
**Pull request by jogibear9988**

### CODE REVIEW

**Comment by jogibear9988** [comment]
and what about this?

File: src/System.Linq.Dynamic.Core/ExpressionParser.cs [hunk 1]
```diff
@@ -181,6 +182,7 @@ interface IEnumerableSignatures
             void LastOrDefault();
             void Single();
             void SingleOrDefault();
+            void ToList();
```
Comment (StefH) [inline src/System.Linq.Dynamic.Core/ExpressionParser.cs]: Is there a test to cover this method?

Comment (StefH) [inline src/System.Linq.Dynamic.Core/ExpressionParser.cs]: And a function like `ToArray`, can that also be defined?

File: test/System.Linq.Dynamic.Core.Tests/DynamicExpressionParserTests.cs [hunk 2]
```diff
@@ -54,6 +54,48 @@ public void ParseLambda_EmptyParameterList()
         }
 
         [Fact]
+        public void ParseLambdaComplex_1()
```
Comment (StefH) [inline test/System.Linq.Dynamic.Core.Tests/DynamicExpressionParserTests.cs]: Is it possible to move this test to the file QueryableTests.GroupBy.cs or does that not fit?

**Comment by StefH** [comment]
I did add some question to this code-review.

File: test/System.Linq.Dynamic.Core.Tests/DynamicExpressionParserTests.cs [hunk 3]
```diff
@@ -54,6 +54,48 @@ public void ParseLambda_EmptyParameterList()
...
+            var expected = qry.GroupBy(x => new { x.Profile.Age }, x => x).OrderBy(gg => gg.Key.Age).Select(j => new { j.Key.Age, TotalIncome = j.Sum(k => k.Income) }).Select(c => new ComplexParseLambda1Result { Age = c.Age, TotalIncome = c.TotalIncome }).Cast<dynamic>().ToArray();
+
+            // Assert
+            Check.That(result).IsNotNull();
+            Check.That(result).HasSize(expected.Length);
+            Check.That(result.ToArray()[0]).Equals(expected[0]);
+        }
+
+        [Fact]
+        public void ParseLambdaComplex_2()
```
Comment (jogibear9988) [inline test/System.Linq.Dynamic.Core.Tests/DynamicExpressionParserTests.cs]: This tests ToList()

File: src/System.Linq.Dynamic.Core/ExpressionParser.cs (hunk 1 above)
Comment (jogibear9988) [reply src/System.Linq.Dynamic.Core/ExpressionParser.cs]: Yes in ParseLambdaComplex_2

Comment (jogibear9988) [reply src/System.Linq.Dynamic.Core/ExpressionParser.cs]: Think so, but I've not tested/used

File: test/System.Linq.Dynamic.Core.Tests/DynamicExpressionParserTests.cs (hunk 2 above)
Comment (jogibear9988) [reply test/System.Linq.Dynamic.Core.Tests/DynamicExpressionParserTests.cs]: The other tests in this class (for example ParseLambda_1) also test Lambda with GroupBy

**Pull request merged**